frontend/src/components/SignupForm.jsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from "react";
+import { Button, Form, Input, message } from "antd";
+import { useAuth } from "../context/AuthContext";
+import { Link, useNavigate } from "react-router-dom";
+
+export default function SignupForm() {
+  const navigate = useNavigate();
+  const [messageApi, contextHolder] = message.useMessage();
+  const { signup } = useAuth();
+  const [loading, setLoading] = useState(false);
+
+  const onFinish = async (values) => {
+    setLoading(true);
+    const success = await signup({
+      username: values.username,
+      email: values.email,
+      password: values.password,
+      firstName: values.firstName,
+      lastName: values.lastName,
+    });
+    setLoading(false);
+    navigate("/login");
+
+    if (success) {
```

Comment:
you do navigation and after that create alert.
better do smth like this
`if (success) {
  messageApi.success("Account created successfully. Please login.");
  setTimeout(() => navigate("/login"), 1500);
} else {
  messageApi.error("Signup failed. Please try again.");
}`